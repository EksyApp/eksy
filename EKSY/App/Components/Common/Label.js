import React, {Component} from 'react'
import {Text, StyleSheet} from 'react-native'
import * as Theme from '../../Theme'
import PropTypes from 'prop-types';
import {StyleShape} from "../../Utils/PropTypeShapes";

export class Label extends Component {
	render() {
		return(
				<Text {...this.props} style={[style.text, this.props.style]} />
		)
	}
}

const style = StyleSheet.create({
	text: {
		textAlign: 'center',
		fontSize: 20,
		color: Theme.accentColor,
		fontWeight: 'bold',
	}
})

export default Label

Label.propTypes = {
	style: StyleShape
}