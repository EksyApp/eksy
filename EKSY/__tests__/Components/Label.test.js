import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Label from "../../App/Components/Common/Label";

test('Label renders correctly', () => {
	const tree = renderer.create(
			<Label />
	).toJSON();
	expect(tree).toMatchSnapshot();
});