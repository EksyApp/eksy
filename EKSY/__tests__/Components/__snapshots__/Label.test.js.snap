// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgb(101,114,122)",
        "fontSize": 20,
        "fontWeight": "bold",
        "textAlign": "center",
      },
      undefined,
    ]
  }
/>
`;
